### PR TITLE
patch(integration_test_charm.yaml): Use controller logsink to generate `juju debug-log`

### LINKED ISSUE #162: `juju debug-log` has line limit on VM
Thanks @Mehdi-Bendriss!

Using `juju debug-log` in integration_test.yaml does not collect all logs

https://bugs.launchpad.net/juju/+bug/2060804

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
Fixes #162

#164 (k8s) is still an issue

Workaround for Juju bug: https://bugs.launchpad.net/juju/+bug/2060804

### CODE REVIEW

**Review by Mehdi-Bendriss — APPROVED** [review]
Thank you!

File: .github/workflows/integration_test_charm.yaml [hunk 1]
```diff
@@ -337,7 +337,18 @@ jobs:
         run: juju status --color --relations | tee ~/logs/juju-status.txt
       - name: juju debug-log
         if: ${{ success() || (failure() && steps.tests.outcome == 'failure') }}
-        run: juju debug-log --color --replay --no-tail | tee ~/logs/juju-debug-log.txt
+        # TODO future improvement: show models/ logs in juju-debug-log.txt to match `juju debug-log`
+        run: |
+          if [[ '${{ inputs.cloud }}' == 'lxd' ]]
+          then
+            mkdir ~/logs/models/
```
Comment (Mehdi-Bendriss) [inline .github/workflows/integration_test_charm.yaml:344]:
```suggestion
            mkdir -p ~/logs/models/
```

File: python/cli/data_platform_workflows_cli/convert_logsink_to_debug_log.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    TRACE = "[39m"
+    DEBUG = "[32m"
+    INFO = "[94m"
+    WARNING = "[33m"
+    ERROR = "[91m"
+    CRITICAL = "[41;97m"
+
+    def __str__(self):
+        """Name with ANSI color code"""
+        return f"{self.value[1]}{self.name}[0m"
```
Comment (Mehdi-Bendriss) [inline python/cli/data_platform_workflows_cli/convert_logsink_to_debug_log.py:26]: In the case of `print(Level.INFO)` what would `9INFO[0m` represent?

**Review by Mehdi-Bendriss — COMMENTED** [review]
Thanks Carl! I have a question

**Review by taurus-forever — APPROVED** [review]
LGTM to unblock you with OpenSearch refresh, but please attach Juju bugreport.
I hope to remove convert-logsink-to-debug-log when juju is fixed and we will return to `juju debug-log`. Tnx!

File: .github/workflows/integration_test_charm.yaml (hunk 1 above)
Comment (phvalguima) [reply .github/workflows/integration_test_charm.yaml:344]: @carlcsaposs-canonical not if `failure() && steps.tests.outcome != 'failure'`

**Pull request merged**